### PR TITLE
Migrate creation and edition component to function component

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Proposed changes

* Migrate creation and edition component to function component
*

### Related issues

*
*

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityEditionOverview.js [hunk 1]
```diff
@@ -242,40 +202,38 @@ class VulnerabilityEditionOverviewComponent extends Component {
...
-      ]),
-    )(vulnerability);
-    return (
+  const createdBy = convertCreatedBy(vulnerability);
+  const objectMarking = convertMarkings(vulnerability);
+  const status = convertStatus(t, vulnerability);
+  const initialValues = pipe(
+    assoc('createdBy', createdBy),
+    assoc('objectMarking', objectMarking),
+    R.assoc('x_opencti_workflow_id', status),
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityEditionOverview.js:219]: why R.assoc here and assoc the line just above?

File: opencti-platform/opencti-front/src/private/components/locations/positions/PositionEditionOverview.js [hunk 2]
```diff
@@ -299,8 +266,8 @@ class PositionEditionOverviewComponent extends Component {
               name="latitude"
               label={t('Latitude')}
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/locations/positions/PositionEditionOverview.js:266]: missing translation for Latitude and Longitude

File: opencti-platform/opencti-front/src/private/components/arsenal/channels/ChannelEditionOverview.js [hunk 3]
```diff
@@ -197,180 +196,168 @@ class ChannelEditionOverviewComponent extends Component {
...
+                    onFocus={handleChangeFocus}
+                    onSubmit={handleSubmitField}
+                    helperText={
+                      <SubscriptionFocus context={context} fieldName="name" />
+                    }
+                  />
+                  <OpenVocabField
+                    type="channel_types_ov"
+                    name="channel_types"
+                    label="Channel types"
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/arsenal/channels/ChannelEditionOverview.js]: translation to add ?

File: opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualEditionOverview.js [hunk 4]
```diff
@@ -1,21 +1,10 @@
...
-  difference,
-  head,
-} from 'ramda';
-import * as Yup from 'yup';
+import React from 'react';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { Field, Form, Formik } from 'formik';
 import * as R from 'ramda';
-import inject18n from '../../../../components/i18n';
+import { assoc, difference, head, map, pathOr, pick, pipe } from 'ramda';
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/entities/individuals/IndividualEditionOverview.js]: It seems that Ramda import are already done line 4

File: opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityEditionOverview.js (hunk 1 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/arsenal/vulnerabilities/VulnerabilityEditionOverview.js:219]: No reason, fixed

File: opencti-platform/opencti-front/src/private/components/locations/positions/PositionEditionOverview.js (hunk 2 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/locations/positions/PositionEditionOverview.js:266]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationEditionOverview.js [hunk 5]
```diff
@@ -1,22 +1,11 @@
...
-  pipe,
-  pick,
-  difference,
-  head,
-} from 'ramda';
+import React from 'react';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { Field, Form, Formik } from 'formik';
+import * as R from 'ramda';
+import { assoc, difference, head, map, pathOr, pick, pipe } from 'ramda';
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationEditionOverview.js]: It is better to import Ramda as you did line 4, then call the different method directly on it : R.assoc, R.head etc. Ramda becomes easier to search this way.

File: opencti-platform/opencti-front/src/private/components/entities/sectors/SectorEditionOverview.js [hunk 6]
```diff
@@ -1,11 +1,10 @@
...
-import { graphql, createFragmentContainer } from 'react-relay';
-import { Formik, Form, Field } from 'formik';
-import { assoc, map, pathOr, pipe, pick, difference, head } from 'ramda';
-import * as Yup from 'yup';
+import React from 'react';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { Field, Form, Formik } from 'formik';
 import * as R from 'ramda';
-import inject18n from '../../../../components/i18n';
+import { assoc, difference, head, map, pathOr, pick, pipe } from 'ramda';
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/entities/sectors/SectorEditionOverview.js]: Same comment as above about Ramda import

File: opencti-platform/opencti-front/src/private/components/entities/systems/SystemEditionOverview.js [hunk 7]
```diff
@@ -1,61 +1,21 @@
...
-  difference,
-  head,
-} from 'ramda';
-import * as Yup from 'yup';
+import React from 'react';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { Field, Form, Formik } from 'formik';
 import * as R from 'ramda';
-import inject18n from '../../../../components/i18n';
+import { assoc, difference, head, map, pathOr, pick, pipe } from 'ramda';
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/entities/systems/SystemEditionOverview.js]: Same comment as above about Ramda

File: opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationEditionOverview.js (hunk 5 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/organizations/OrganizationEditionOverview.js]: Very good point
Fixed

File: opencti-platform/opencti-front/src/private/components/arsenal/channels/ChannelEditionOverview.js (hunk 3 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/arsenal/channels/ChannelEditionOverview.js]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/sectors/SectorEditionOverview.js (hunk 6 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/sectors/SectorEditionOverview.js]: Fixed

File: opencti-platform/opencti-front/src/private/components/entities/systems/SystemEditionOverview.js (hunk 7 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/entities/systems/SystemEditionOverview.js]: Fixed

**Comment by Archidoit** [comment]
Bugs found:
- when editing an Opinion: the initial values for Description is always empty even if the opinion has a description.
 (updated by Romuald) Fixed

- Event edition: when modifying Author , get an error : 
image Fixed

- is it normal to have an explanation for Attack patterns without a description (this behavior is not the same in other entities) : 
image

**Comment by marieflorescontact** [comment]
**Bugs found:**

- Can't update the details part of a `Malware` with external references enabled
-> (updated by Romuald) Seems to be a generic problem, we need to create an issue [issues 2894](https://github.com/OpenCTI-Platform/opencti/issues/2894)

- When creating a `vulnerability`, score and severity shouldn't be select elements? Now we can give numbers wich is forbidden for severity and gives an error for score:

Capture d'écran_20230216_120256
(updated by Romuald) Fixed

- Can't update the threat actor types field of a `threat actor`
(updated by Romuald) Fixed

**Question:**

- In `Analysis/Reports`, when clicking on a report, we go directly on Knowledge section (we usually go to Overview).
Is it normal?

**Comment by RomuDeuxfois** [comment]
@SamuelHassine & @richard-julien, after reviewing and testing this pull request, one question remain unanswered :

- is it normal to have an explanation for Attack patterns without a description (this behavior is not the same in other entities)
image
(see Archidoit comment)

**Comment by Archidoit** [comment]
@RomuDeuxfois to respond to the comment of @marieflorescontact about the redirection in Report : the redirection depends on the redirectionMode that is selected (you can change the redirectionMode via the buttons : 
image
)

**Review by richard-julien — APPROVED** [review]

**Pull request merged**